pkg/pillar/containerd/containerd.go

```diff
@@ -940,3 +1020,27 @@ func bindNS(ns string, path string, pid int) error {
 	}
 	return nil
 }
+func newServiceCtx(namespace string) (context.Context, context.CancelFunc) {
+	return context.WithCancel(namespaces.WithNamespace(context.Background(), namespace))
+}
+
+func newServiceCtxWithLease(ctrdClient *containerd.Client, namespace string) (context.Context, context.CancelFunc, error) {
+	if ctrdClient == nil {
+		return nil, nil, fmt.Errorf("newServiceCtxWithLease(%s): exception while verifying ctrd client: "+
+			namespace, "Container client is nil")
+	}
+	ctx, cancle := newServiceCtx(namespace)
```

Comment:
Typo

Reply:
Why do you need two levels of WithCancel and two cancel functions?

Reply:
Corrected the typo.
Calling the done() returned from `ctx, done, err := ctrdClient.WithLease(ctx)` just deletes the lease as fas as I saw, it didn't cancel the context. Because of that, I'm cancelling the context separately after deleting the context.

Reply:
OK. How about adding a comment about that.

Reply:
Done.